src/components/space/satellite.tsx

```diff
@@ -47,17 +49,23 @@ export const Satellite = ({ color, tle }: SatelliteProps) => {
 
   camera.getWorldDirection(vector);
 
-  return (
-    <>
+  if (!visibility) {
+    return null;
+  }
+  else {
+    return (
+      <>
       <mesh ref={planeteRef}>
-        <sphereGeometry args={[0.1, 8, 8]} />
-        <meshBasicMaterial color={color} />
-      </mesh>
-      <mesh ref={textRef}>
-        <Text color={color} outlineColor={0x000000} outlineWidth={0.01}>
-          {getSatelliteName(tle)}
-        </Text>
-      </mesh>
-    </>
-  );
+          <sphereGeometry args={[0.1, 8, 8]} />
+          <meshBasicMaterial color={color} />
+        </mesh>
+        <mesh ref={textRef}>
+          <Text color={color} outlineColor={0x000000} outlineWidth={0.01}>
+            {getSatelliteName(tle)}
+          </Text>
+        </mesh>
+      </>
+    );
+  }
+  
```

Comment:
React Three Fiber components like `<mesh>` have a `visible` prop that can be used instead of early return.